File: 2023/Primeira Fase/Subsequência/subsequencia.py
a, b = [int(x) for x in input().split()]

sa = [int(x) for x in input().split()]

sb = [int(x) for x in input().split()]

pont_a = 0
pont_b = 0

while pont_a<a and pont_b<b:

    if sa[pont_a] == sb[pont_b]:
        pont_b+=1

    pont_a+=1

if pont_b == b:
    print('S')
else:
    print('N')

File: 2023/Primeira Fase/Subsequência/subsequencia.cpp
#include <bits/stdc++.h>

using namespace std;

int a, b, sa[100005], sb[100005];

int main(){
    cin >> a >> b;

    for(int i=0; i<a; i++){
        cin >> sa[i];
    }

    for(int i=0; i<b; i++){
        cin >> sb[i];
    }

    int pont_a = 0;
    int pont_b = 0;

    while(pont_a<a && pont_b<b){
        
        if(sa[pont_a] == sb[pont_b]){
            pont_b++;
        }
        pont_a++;
    }

    if(pont_b == b){
        cout << 'S' << endl;
    }else{
        cout << 'N' << endl;
    }
    return 0;
}
